sycl/doc/extensions/experimental/sycl_ext_intel_runtime_buffer_location.asciidoc

```diff
@@ -0,0 +1,119 @@
+= SYCL_INTEL_usm_runtime_properties
+
+== Introduction
+IMPORTANT: This specification is a draft.
+
+NOTE: Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by permission by Khronos.
+
+NOTE: This document is better viewed when rendered as html with asciidoctor.  GitHub does not render image icons.
+
+NOTE: This serves as a temporary workaround for usm allocation to accept buffer location properties, and the formal solution may change in the future.
+
+This document propose a new buffer_location runtime property that can be passed to `malloc_device`.
+
+== Name Strings
+
++SYCL_INTEL_runtime_buffer_location+
+
+== Contributors
+Aditi Kumaraswamy, Intel +
+Gregory Lueck, Intel +
+Joe Garvey, Intel +
+Sherry Yuan, Intel +
+Steffen Larsen, Intel
+
+== Notice
+
+Copyright (c) 2020 Intel Corporation.  All rights reserved.
+
+== Status
+
+Working Draft
+
+This is a preview extension specification, intended to provide early access to a feature for review and community feedback. When the feature matures, this specification may be released as a formal extension.
+
+Because the interfaces defined by this specification are not final and are subject to change they are not intended to be used by shipping software products.
+
+== Version
+
+Built On: {docdate} +
+Revision: 1
+
+== Overview
+
+This spec describes the solution of passing runtime buffer location properties to malloc APIs, specifically `malloc_device`.
+
+On targets that provide more than one type of global memory, this provide users the flexibility of choosing which memory the device usm should be allocated to.
+
+This information is not a hint; it is a functional requirement of the program that must be respected.
```

Comment:
I think I started a conversation about this someplace else too ... It would be better if this extension had some defined behavior when the device is not FPGA.  It seems like "buffer location" is a concept that only makes sense on FPGA-type devices.  In this case, it seems like it would make sense to say that the `buffer_location` is allowed when allocating memory for any device, but it only has an effect when the device is an FPGA.

Reply:
Thanks I mentioned it in the [Note] section

Reply:
I'd suggest removing this last sentence.  This is only the overview section.  There should be a more formal definition of the property below in the "Specification" section.